xarray/plot/plot.py

```diff
@@ -240,6 +240,47 @@ def _infer_line_data(darray, x, y, hue):
     return xplt, yplt, hueplt, xlabel, ylabel, huelabel
 
 
+def _infer_scatter_data(ds, x, y, hue):
+    dvars = set(ds.data_vars.keys())
+    error_msg = (' must be either one of ({0:s})'
+                 .format(', '.join(dvars)))
+
+    if x not in dvars:
+        raise ValueError(x + error_msg)
+
+    if y not in dvars:
+        raise ValueError(y + error_msg)
+
+    dims = ds[x].dims
+    if ds[y].dims != dims:
+        raise ValueError('{} and {} must have the same dimensions.'
+                         ''.format(x, y))
+
+    if hue is not None and hue not in dims:
```

Comment:
Should think of multidimensional co-ordinates here. 

Example:
```
ds = xr.tutorial.load_dataset('rasm')

ds['T2'] = ds.Tair**2
ds.scatter(x='Tair', y='T2', hue='xc')
```

Reply:
How can this be done? `xc` is not a one dimensional object with a small set of values it takes, it can be arbitrary. For all we know, each entry in `Tair` can have a different `xc`. So that means we'll have a very long legend? Or were you thinking about a "continuous legend", i.e. a colorbar?

Reply:
> xc is not a one dimensional object with a small set of values it takes, it can be arbitrary. 

This is a problem even for long 1D co-ordinates i.e. you could have a huge, useless legend. How about:
1) `ds.scatter(x='x', y='y', hue='z', add_legend=True)` puts up a legend as in the examples above.

2) `ds.scatter(x='x', y='y', hue='z', add_colorbar=True)` uses a colorbar instead of a legend?

Option 2 could be the sane default. It's also useful because sometimes you want increasingly-darker (or lighter) shades to represent monotonic variation of the `hue` variable even when the `hue` variable has only 3 or 4 levels.

`add_colorbar` and `add_legend` would be mutually exclusive.